### PR TITLE
Implemented IDisposable interface

### PR DESCRIPTION
**Pull request by khanova**
Implemented IDisposable interface for classes:
* CatBoostModel
* CatBoostModelEvaluator

### CODE REVIEW

File: catboost/dotnet/CatBoostNet/CatBoostModelEvaluator.cs [hunk 1]
```diff
@@ -185,5 +187,28 @@ private struct CatBoostModelContainer {
...
+        public void Dispose() {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Dispose of unmanaged resources
+        /// </summary>
+        protected virtual void Dispose(bool fromDisposeMethod) {
+            if (isDisposed) {
```
Comment (kizill) [inline catboost/dotnet/CatBoostNet/CatBoostModelEvaluator.cs]: code will never enter this branch

File: catboost/dotnet/CatBoostNet/CatBoostModel.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ public class CatBoostValuePrediction
     /// If you need to implement a custom model code, use <see cref="CatBoostModelEvaluator"/>
     /// as the basic building block.
     /// </summary>
-    public class CatBoostModel : ITransformer
+    public class CatBoostModel : ITransformer, IDisposable
```
Comment (kizill) [inline catboost/dotnet/CatBoostNet/CatBoostModel.cs]: you inherit from IDisposable but do not write Dispose method

File: catboost/dotnet/CatBoostNet/CatBoostModelEvaluator.cs (hunk 1 above)
Comment (khanova) [reply catboost/dotnet/CatBoostNet/CatBoostModelEvaluator.cs]: Fixed

File: catboost/dotnet/CatBoostNet/CatBoostModel.cs (hunk 2 above)
Comment (khanova) [reply catboost/dotnet/CatBoostNet/CatBoostModel.cs]: Discussed

**Comment by kizill** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@khanova, please agree to the CLA (https://github.com/catboost/catboost/blob/master/CONTRIBUTING.md#provide-contributions)

**Comment by arcadia-devtools** [comment]
@kizill, internal review request created: [1200143](https://a.yandex-team.ru/review/1200143)

**Comment by arcadia-devtools** [comment]
Check [1200143](https://a.yandex-team.ru/review/1200143) passed

**Comment by khanova** [comment]
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=ru

**Pull request merged**